### PR TITLE
Feature/divider

### PR DESCRIPTION
**Pull request by ShaunDonn**
Creates a button in dashboard called "Divide Personnel" that, when pressed, displays a table of the Personnel who work in all warehouses. I don't love the way it flows or the fact that the button is called "Divide Personnel", but for the sake of meeting the rubric requirements, it's enough for now. Would like to add the button to the Personnel page if there's time.

Created a new class, the same as accounts/personnel, but it only includes the Personnel id, first_name, and last_name. 

personnel_divided.ts has a method called listDividedPersonnel, which returns the result of the division query. This method is called in dividerRoute, which gives it the name "div_result". This variable is then used to fill in the table body in divider.ejs.

I know the naming isn't very straight forward, so if you guys have any suggestions for changes to names of any classes, variables, etc. let me know.

### CODE REVIEW

File: web/src/model/division/personnel_divided.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export default class DividedPersonnel {
```
Comment (pseigo) [inline web/src/model/division/personnel_divided.ts:1]: DividedPersonnel or PersonnelDivided?

File: web/src/model/division_personnel.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                        (SELECT DISTINCT N.warehouse_id
+                        FROM works_in N
+                        WHERE N.pid = P.id));`,
+        values: [],
+      };
+
+      let result = this.db.client.query(query)
+      .then(res => {
+        return res.rows.map(p => {
+          return new DividedPersonnel(p.id, p.first_name, p.last_name);
```
Comment (pseigo) [inline web/src/model/division_personnel.ts:29]: You should be able to re-use the Personnel class, no? This seems like duplication to me

File: web/src/routes/divider/dividerRoute.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        this.personnel_division = new PersonnelDivide(db);
+        this.initializeRoutes();
+    }
+
+    private initializeRoutes(): void {
+    
+        this.router.get("/", (req, res) => {
+            this.personnel_division.listDividedPersonnel()
+            .then(div_result => {
+                res.render('web/page/divider/divider.ejs', {title: "Divided Personnel", div_result: div_result});
```
Comment (pseigo) [inline web/src/routes/divider/dividerRoute.ts:23]: Would probably want to rename that template to `index.ejs` for consistency :)

**Review by pseigo — CHANGES_REQUESTED** [review]
Minor changes, otherwise looks good!

**Review by pseigo — APPROVED** [review]

**Comment by pseigo** [comment]
Ready to merge! (make sure to reinit your DB after this gets merged @nuxeric )

**Comment by pseigo** [comment]
Just making it easy for Adi to check off the boxes :)

image
image

File: sql_script/initialize_db.sql [hunk 4]
```diff
@@ -1,22 +1,45 @@
-CREATE TABLE personnel ( 
-    id INTEGER, 
-    birth_date DATE, 
-    first_name TEXT, 
-    last_name TEXT, 
+DROP TABLE categorizes CASCADE;
```
Comment (pseigo) [inline sql_script/initialize_db.sql:1]: This is the only change here, other lines are just whitespace changes :P

**Comment by ShaunDonn** [comment]
Nice changes. Thanks @pseigo!

**Review by nuxeric — APPROVED** [review]

**Pull request merged**